### DISCUSSION TITLE
Data provider filter via a custom query

### DISCUSSION THREAD

**Discussion by Mozix** [Budibase/budibase#12192] [Help] [answered]
Hey,

How can i filter my table that comes from a custom query?

Let's say I have a page1 when I choose an option from the table in page1, I wanna filter the data in page2 based on my choice in page1 how can I approach this with custom queries?

Thanks!

**Comment by ConorWebb96** [comment]
Hey @Mozix,

It sounds like you want to persist a value through [state](https://docs.budibase.com/docs/app-state#persisting-values). Then use this saved value as the binding of your query on the second page. I've linked the documentation above for persisting/state in general. 

I'm assuming from what you've written that you already know how to create a binding for a custom query. All thats left is to save the second value to state with persist checked whenever a row is clicked. Then reference the created state in your data provider settings `cog`, custom query area of the second page and you should be able to see the results of the second query.

If you need an example after this explanation, let me know and I'll do one up for you.

I hope this explanation helps.

**Reply by Mozix** [reply]
Hey @ConorWebb96

I have managed to create the state but I am stuck at updating the query part I don't seem to get that part so an example would be amazing!

Thank you again!

**Reply by ConorWebb96** [reply, marked as answer]
Hey @Mozix,

Sorry for the delay on responding to this one. Below are the steps I used to create a few simple exmaple.

1. Created a query in my case called `Pet Type`.
2. Simple select from the pets table to get the type of animal.
<img width="1286" alt="Screenshot 2023-10-30 at 07 46 07" src="https://github.com/Budibase/budibase/assets/126772285/35cd7041-c117-4894-9a93-fde647d00482">
3. Created 2 pages one for the entire pets table and the other for queries results.
<img width="1512" alt="Screenshot 2023-10-30 at 07 46 41" src="https://github.com/Budibase/budibase/assets/126772285/d3d64ca8-da26-476d-b631-074131cd5744">
4. Added a create state action to a table based on the clicked row. Saved the type to state from the clicked row.
<img width="1512" alt="Screenshot 2023-10-30 at 07 47 07" src="https://github.com/Budibase/budibase/assets/126772285/0049eb96-957a-4b90-a30f-35c5059c45b8">
5. Referenced the persisted state inside the query settings area of the data provider within the table block.
<img width="1511" alt="Screenshot 2023-10-30 at 07 47 51" src="https://github.com/Budibase/budibase/assets/126772285/93353798-3751-49ab-bbad-0bd4ec7468a8">

**End result**
![Screen Recording 2023-10-30 at 07 50 22](https://github.com/Budibase/budibase/assets/126772285/b444a77c-870d-4058-8316-4b29396460f2)

I hope this helps.

**Reply by cl-provecto** [reply]
Would you help me? Using this same example, how would I use a field for Date
